### PR TITLE
feature: add sustainable web design model

### PR DESCRIPTION
**Pull request by drydenwilliams**
PR for issue #56

https://github.com/thegreenwebfoundation/co2.js/issues/56

### Acceptance criteria:

- [x] Link to where KWH_PER_GB figure comes from in code comment.
- [x] Fix all GLOABL_INTENSITY typos
- [x] Link to underlying source of carbon intensity info figure of 475, in being explicit about the year in use in code comment
- [x] replace 0.52 magic number with constant explaining what it represents i.e `END_USER_DEVICE_ENERGY` 
- [x] fix `annualSegmentEngergy` typo
- [x] fix order of variables in energy per visit method, and reformat for more readability as described

### CODE REVIEW

**Comment by mrchrisadams** [comment]
Thanks for this @drydenwilliams.

I've added a few comments and requests for changes - mainly code comments and a typo TBH. If you make those changes I'm happy to merge these in, as it's a definite improvement on what we have right now 👍

**Comment by mrchrisadams** [comment]
Here's the issue for other bits that came up that I think are worth capturing, but not trying to address in this PR. this woudl make sense to use in a future version

https://github.com/thegreenwebfoundation/co2.js/issues/60

File: src/sustainable-web-design.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+  annualEnergyInKwh(energyPerVisit, monthlyVisitors = 1000) {
+    return energyPerVisit * monthlyVisitors * 12;
+  }
+
+  annualEmissionsInGrams(co2grams, monthlyVisitors = 1000) {
+    return co2grams * monthlyVisitors * 12;
+  }
+
+  annualSegmentEngergy(annualEnergy) {
```
Comment (mrchrisadams) [inline src/sustainable-web-design.js:48]: small typo here ~~`annualSegmentEngergy`~~ - `annualSegmentEnergy`

File: src/sustainable-web-design.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+class SustainableWebDesign {
+  constructor(options) {
+    this.options = options;
+  }
+
+  energyPerVisit(bytes) {
+    const transferedBytesToGb = bytes / fileSize.GIGABYTE;
+    return (
+      transferedBytesToGb * KWH_PER_GB * RETURNING_VISITOR_PERCENTAGE +
```
Comment (mrchrisadams) [inline src/sustainable-web-design.js:28]: Given this is representing two separate things, why not make this more explicit with named variables for two separate things?

I can't speak for others, but I think it makes it easier to see errors being introduced when during implementation.

I think this is intended to be roughly equivalent to _Energy per visit in kWh (E)_ part of the model from [the SWD page](https://sustainablewebdesign.org/calculating-digital-emissions/). I've broken it onto separate lines to make it a tad easier to read:

> (Data Transfer per Visit (new visitors) in GB x 0.81kWh/GB x 0.75) 
> + 
> (Data Transfer per Visit (returning visitors) in GB x 0.81kWh/GB x 0.25 x 0.02)
> 
> 

The code in this PR here has `FIRST_TIME_VIEWING_PERCENTAGE` and `RETURNING_VISITOR_PERCENTAGE` the other way around when compared to the initial model.

I also missed it the first time when reviewing this PR 😅

**Review by mrchrisadams — CHANGES_REQUESTED** [review]
Hi Dryden, thanks for making this - there's a few small changes to make.

**Comment by drydenwilliams** [comment]
Thanks @mrchrisadams will take a look through this hopefully this week, thanks for taking the time again!

**Comment by mrchrisadams** [comment]
Thanks for this Dryden!

I'll merge it in now, and make a bit of time to update the README file as outlined in #60, and make any further tweaks needed to get it ready for publishing a new release.

**Pull request merged**